src/Workleap.DomainEventPropagation.Subscription/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -7,17 +7,32 @@ namespace Workleap.DomainEventPropagation.Extensions;
 [ExcludeFromCodeCoverage]
 public static class ServiceCollectionEventPropagationExtensions
 {
-    public static IEventPropagationSubscriberBuilder AddEventPropagationSubscriber(this IServiceCollection services, Action<EventPropagationSubscriberOptions> configureOptions = null)
+    public static IEventPropagationSubscriberBuilder AddEventPropagationSubscriber(this IServiceCollection services)
+        => services.AddEventPropagationSubscriber(_ => { });
+
+    public static IEventPropagationSubscriberBuilder AddEventPropagationSubscriber(this IServiceCollection services, Action<EventPropagationSubscriberOptions> configure)
     {
+        if (services == null)
+        {
+            throw new ArgumentNullException(nameof(services));
+        }
+
+        if (configure == null)
```

Comment:
If there isn't one already we could add a test class for this and unit test those checks and also make sure the action still works as expected

Reply:
I've added both unit and integration tests to test various scenarios, but mostly guards and options. 🧪